include/deal.II/grid/grid_tools.h

```diff
@@ -556,6 +556,100 @@ namespace GridTools
   get_active_neighbors (const typename Container::active_cell_iterator        &cell,
                         std::vector<typename Container::active_cell_iterator> &active_neighbors);
 
+  /**
+   * Extract and return the active cell layer around a subdomain (set of active
+   * cells) in the @p container (i.e. those that share a common set of vertices
+   * with the subdomain but are not a part of it).
+   * Here, the "subdomain" consists of exactly all of those cells for which the
+   * @p predicate returns @p true.
+   *
+   * @note This function may return both locally owned and ghost cells, but
+   * is expected not to return artificial cells.
+   *
+   * @tparam Container A type that satisfies the requirements of a mesh
+   * container (see @ref GlossMeshAsAContainer).
+   * @param[in] container A mesh container (i.e. objects of type Triangulation,
+   * DoFHandler, or hp::DoFHandler).
+   * @param[in] predicate A function defining the subdomain around which the
+   * halo layer is to be extracted. It is a function that takes in an active cell
+   * and returns a boolean.
```

Comment:
I think this would be a good place to say something like "Examples of predicates that are frequently useful can be found in namespace IteratorFilters."